polygeist/lib/Dialect/Polygeist/Transforms/SYCLRaiseHost.cpp

```diff
@@ -118,7 +302,9 @@ void SYCLRaiseHostConstructsPass::runOnOperation() {
   MLIRContext *context = &getContext();
 
   RewritePatternSet rewritePatterns{context};
-  rewritePatterns.add<RaiseKernelName>(context);
+  // RaiseKernelName should be prioritized, as RaiseSetKernel depends on that.
```

Comment:
Is the order here only relevant for efficiency or for correctness?

Reply:
Nope, `benefit` handles the correctness (introduces priority)

Reply:
Sorry, should have been clearer: Is the priority important for correctness or just for efficiency with a greedy driver?

Reply:
It is important for correctness, as we must raise `sycl.host.get_kernel` before running the other pattern, which will fail otherwise